group_project_v2/mixins.py

```diff
@@ -266,18 +266,30 @@ class DashboardRootXBlockMixin(ProjectAPIXBlockMixin):
     TARGET_STUDENTS = 'target_students'
     TARGET_WORKGROUPS = 'target_workgroups'
 
-    def _append_context_parameters_if_not_present(self, context):
+    FILTERED_STUDENTS = "filtered_students"
+
+    def _add_students_and_workgroups_to_context(self, context, filter_by_client_id=None):
         """
-        Appends target students and target workgroups parameters, if not already present in context
-        :param dict context: XBlock view context
-        :rtype: None
+        :param dict context:
+        :param int filter_by_client_id: If none we won't do filtering
+        :return:
         """
-        workgroups, users = self.get_workgroups_and_students()
-        if self.TARGET_STUDENTS not in context:
-            context[self.TARGET_STUDENTS] = users
+        workgroups, students = self.get_workgroups_and_students()
+
+        context[self.TARGET_STUDENTS] = list(students)
+        context[self.TARGET_WORKGROUPS] = list(workgroups)
+        context[self.FILTERED_STUDENTS] = filtered_students = set()
```

Comment:
@jbzdak nit: chained assignment - technically nothing wrong with it, but some people just don't like it (for various reasons, like having experience with languages that don't allow it). I'm one of those people, so could you please expand this to two assignments?